scripts/cc-cli/src/utils/tx.ts

```diff
@@ -28,21 +28,28 @@ export async function signSendAndWatch(
         unsubAndResolve(result);
       }
       if (dispatchError) {
+        let blockHash: string | null = null;
+        if (status.isInBlock) blockHash = status.asInBlock.toHex();
+
         if (dispatchError.isModule) {
           // for module errors, the section is indexed, lookup
           const decoded = api.registry.findMetaError(dispatchError.asModule);
           const { docs, name, section } = decoded;
           const error = `${section}.${name}: ${docs.join(" ")}`;
           const result = {
             status: TxStatus.failed,
-            info: `Transaction failed with error: "${error}"`,
+            info: `Transaction failed with error: "${error}" ${
+              blockHash ? "at block " + blockHash : ""
```

Comment:
Q2: What are the steps to reproduce here ? How do I get to cause a failure? 

Or do you mean any failure whatsoever ?

Reply:
its harder to test it, because you need to comment the check balance before, and also many of our CLI ensure that the transaction has the most chance to succeed, but since I haven't merged the wizard yet in the other PR.

You can actually faucet 50000 CTC and wizard exactly 50000 CTC causing it to fail on chain.

Reply:
If I understand this correctly then the steps to reproduce are: 

1) Call `wizard` and use the same seed for Stash & Controller accounts AND
2) Specify the maximum available amount to bond.

I got the following

```
$ node dist/index.js balance  -u ws://127.0.0.1:9946 -a 5FEK97W73jQHWdUYAeZvyb7iRuygZhFfC75ZA9cfqkQ8yhvb
Available: 1128797.333449876543955951 CTC
Free: 1128797.333449876543955951 CTC
Reserved: 0.000000000000000000 CTC
Misc Frozen: 0.000000000000000000 CTC
Fee Frozen: 0.000000000000000000 CTC


$ node dist/index.js wizard  -u ws://127.0.0.1:9946 -a 1128797.333449876543955951
🧙 Running staking wizard...
✔ Specify a seed phrase for the Stash account … **************************************************************************
✔ Specify a seed phrase for the Controller account … **************************************************************************
Using the following parameters:
💰 Stash account: 5FEK97W73jQHWdUYAeZvyb7iRuygZhFfC75ZA9cfqkQ8yhvb
🎮 Controller account: 5FEK97W73jQHWdUYAeZvyb7iRuygZhFfC75ZA9cfqkQ8yhvb
🪙 Amount to bond: 1128797.333449876543955951 CTC
🎁 Reward destination: Staked
📡 Node URL: ws://127.0.0.1:9946
💸 Commission: 0
🔐 Blocked: No
✔ Continue? … yes
Sending bond transaction...
Sending transaction...
Transaction included at blockHash 0xd64d9296b18ecb7392b6babe528bc17e895c2859dacb55e68fcc88faae6f994b
Generating new session keys on node...
New node session keys: 0xe92f9e747a67c9d2735b6569cfc8a45fff00d6e1ac2010918e9d419f60b50db7a6f7ae0ec965f76b8e780027615997438848ebcabd425cff0dec5c324180bd00bc8ca6e01c4c3ce041143d31a7f805691cd7323ed79d534840d28fb2fe402210
Creating setKeys transaction...
Creating validate transaction...
Sending setKeys and validate transactions...
Sending transaction...
2023-06-24 00:07:21        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: Inability to pay some fees , e.g. account balance too low
2023-06-24 00:07:21        RPC-CORE: submitAndWatchExtrinsic(extrinsic: Extrinsic): ExtrinsicStatus:: 1010: Invalid Transaction: Inability to pay some fees , e.g. account balance too low
/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/@polkadot/rpc-provider/cjs/coder/index.js:30
    throw new _error.default(`${code}: ${message}${formatErrorData(data)}`, code, data);
          ^

RpcError: 1010: Invalid Transaction: Inability to pay some fees , e.g. account balance too low
    at checkError (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/@polkadot/rpc-provider/cjs/coder/index.js:30:11)
    at RpcCoder.decodeResponse (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/@polkadot/rpc-provider/cjs/coder/index.js:45:5)
    at WsProvider.value (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/@polkadot/rpc-provider/cjs/ws/index.js:247:90)
    at W3CWebSocket.value [as onmessage] (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/@polkadot/rpc-provider/cjs/ws/index.js:230:153)
    at W3CWebSocket._dispatchEvent [as dispatchEvent] (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/yaeti/lib/EventTarget.js:107:17)
    at W3CWebSocket.onMessage (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/websocket/lib/W3CWebSocket.js:234:14)
    at WebSocketConnection.<anonymous> (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/websocket/lib/W3CWebSocket.js:205:19)
    at WebSocketConnection.emit (node:events:527:28)
    at WebSocketConnection.processFrame (/home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/websocket/lib/WebSocketConnection.js:554:26)
    at /home/gluwa/creditcoin-2.0/creditcoin-substrate/scripts/cc-cli/node_modules/websocket/lib/WebSocketConnection.js:323:40

Node.js v18.1.0
```

Reply:
If you cause a failure tho, we've got something to patch. Not sure how to easily test failed transactions. Should we add a method to skip check? Like `--unsafe`?